p04-mui-quotes/src/Components/Quote/Quotes.js

```diff
@@ -1,42 +1,146 @@
+
+
 import { Grid, IconButton, Typography } from '@mui/material'
 import React from 'react'
 import CurrentQuote from './CurrentQuote'
-
+import axios from "axios"
 import { Favorite, SkipNext, FavoriteBorderOutlined } from '@mui/icons-material';
-
 import { QuoteActions, QuoteContainer, QuoteListContainer } from './Quotes.styles'
-import QuotesFavsList from './QuotesFavsList';
+import QuotesFavsList from './Favorites/QuotesFavsList';
+import ListTags from './Tags/ListTags';
+import ListFavoriteQuote from './Favorites/ListFavoriteQuote';
+
+
+import {useRef, useState, useEffect} from 'react'
+import _defaultMaxTime, {_APIQuotes} from './QuotesConfig'
+
+
+
 
 const Quotes = () => {
+ 
+  const [counter, setCounter] = useState(_defaultMaxTime)
+  const [counterActive, setCounterActive] = useState(true)
+
+  const [quote, setQuote] = useState([])
+  const [favorite, setFavorite] = useState(false)
+
+  const getQuote = () =>{
+    axios.get(_APIQuotes)
+    .then(res => {
+      setQuote(res)
+    }).catch(err => {
+      console.log(err)
+    })
+  }
+
+  const nextQuote = () =>{
+    setCounter(_defaultMaxTime) 
+    setCounterActive(true)
+    setFavorite(false)
+    getQuote()
+    return quote
+  }
+
+  const [favoritesQuotes, setFavoritesQuotes] = useState([])
+
+  const reduceCounter = () => {
+    (counter - 1)>0 ? setCounter(counter - 1) : nextQuote()
+  }
+
+  useEffect(() => {
+    const timer =
+      (counter > 0 && counterActive)&& setInterval(() => reduceCounter(), 1000);
+      
+      
+    return () => {
+      clearInterval(timer);
+    }
+  }, [counter]);
+
+  useEffect(()=>{
+    nextQuote()
+  },[])
+
+
+  const handleNext = () => {
+    nextQuote()
+  }
+
+  const handlePause = () => {
+    setCounterActive(!counterActive)
+    if (!counterActive) reduceCounter()
+  }
+
+  function addToFavorites () {
+
+    let exist = favoritesQuotes.find((element) => element.id === quote.data._id)
+
+    if (!exist){
+      setFavoritesQuotes(favoritesQuotes => [...favoritesQuotes, 
+        {
+          content : quote.data.content,
+          author: quote.data.author,
+          id : quote.data._id
+        }
+      ])
+    }
+  }
+
+  function removeFromFavorites (key){
+    //console.log(`Removiendo ${key}`)
+    setFavoritesQuotes( (current) =>
+      current.filter((quote) => quote.id !== key)
+    );
+
+    //console.log (`quote.id = ${quote.data._id} and key=${key}`)
+    if (quote.data._id === key) { setFavorite(false)}
```

Comment:
just a tip, if you have one action after the if statement, you can if you like omit the brackets. Example

```
if ( quote.data._id === key ) setFavorite(false)
```

depending on the eslint configuration it may require to be with a break line or not.